### PR TITLE
PUB-1564 Spring Updates

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1564

### Change description ###

- Updated Spring boot to latest (2.7.4)
- Removed Spring fox and replaced it with Spring Doc
- Updated JUnit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -95,26 +95,26 @@
   </suppress>
   <suppress>
     <notes><![CDATA[
-   file name: spring-cloud-azure-starter-active-directory-4.0.0.jar
+   file name: spring-cloud-azure-starter-active-directory-4.4.0.jar
```
Comment (ChrisS1512) [inline config/owasp/suppressions.xml:98]: If you've not already, could you have a quick look to see if any of the above can be removed now that it's upgraded?

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (joshblackmoor) [reply config/owasp/suppressions.xml:12]: Good point. Have removed the ones we no longer need :)

**Review by ChrisS1512 — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -170,37 +170,39 @@ dependencies {
...
   testImplementation group: 'org.springframework.boot', name: 'spring-boot-starter-test', {
     exclude group: 'junit', module: 'junit'
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
   implementation 'org.apache.tomcat.embed:tomcat-embed-core:9.0.63'
   implementation 'org.apache.tomcat.embed:tomcat-embed-websocket:9.0.63'
 
+  testImplementation group: 'org.apiguardian', name: 'apiguardian-api', version: '1.1.2'
+
   testImplementation group: 'com.h2database', name: 'h2', version: '2.1.210'
```
Comment (furnivall) [inline build.gradle:206]: Just spotted we're still using h2 on subs man. Need to make a ticket to move this over to zonky embeddeddatabase like the others. Futureproofs us on this.

**Review by furnivall — APPROVED** [review]
Looks really good dude. Well done for doing this.

File: build.gradle (hunk 2 above)
Comment (joshblackmoor) [reply build.gradle:206]: Just raised a ticket for this :) https://tools.hmcts.net/jira/browse/PUB-1585

**Pull request merged**